src/order/filter/basic.lean

```diff
@@ -1446,6 +1446,69 @@ lemma tendsto_at_top [preorder β] (m : α → β) (f : filter α) :
   tendsto m f at_top ↔ (∀b, {a | b ≤ m a} ∈ f) :=
 by simp only [at_top, tendsto_infi, tendsto_principal]; refl
 
+lemma tendsto_at_top_mono' [preorder β] (l : filter α) ⦃f₁ f₂ : α → β⦄ (h : {x | f₁ x ≤ f₂ x} ∈ l) :
+  tendsto f₁ l at_top → tendsto f₂ l at_top :=
+assume h₁, (tendsto_at_top _ _).2 $ λ b, mp_sets ((tendsto_at_top _ _).1 h₁ b)
+  (monotone_mem_sets (λ a ha ha₁, le_trans ha₁ ha) h)
+
+lemma tendsto_at_top_mono [preorder β] (l : filter α) :
+  monotone (λ f : α → β, tendsto f l at_top) :=
+λ f₁ f₂ h, tendsto_at_top_mono' l $ univ_mem_sets' h
+
+section ordered_monoid
+
+variables [ordered_cancel_comm_monoid β] (l : filter α) {f g : α → β}
+
+lemma tendsto_at_top_add_nonneg_left' (hf : {x | 0 ≤ f x} ∈ l) (hg : tendsto g l at_top) :
+  tendsto (λ x, f x + g x) l at_top :=
+tendsto_at_top_mono' l (monotone_mem_sets (λ x, le_add_of_nonneg_left) hf) hg
+
+lemma tendsto_at_top_add_nonneg_left (hf : ∀ x, 0 ≤ f x) (hg : tendsto g l at_top) :
+  tendsto (λ x, f x + g x) l at_top :=
+tendsto_at_top_add_nonneg_left' l (univ_mem_sets' hf) hg
+
+lemma tendsto_at_top_add_nonneg_right' (hf : tendsto f l at_top) (hg : {x | 0 ≤ g x} ∈ l) :
+  tendsto (λ x, f x + g x) l at_top :=
+tendsto_at_top_mono' l (monotone_mem_sets (λ x, le_add_of_nonneg_right) hg) hf
+
+lemma tendsto_at_top_add_nonneg_right (hf : tendsto f l at_top) (hg : ∀ x, 0 ≤ g x) :
+  tendsto (λ x, f x + g x) l at_top :=
+tendsto_at_top_add_nonneg_right' l hf (univ_mem_sets' hg)
+
+lemma tendsto_at_top_of_add_const_left (C : β) (hf : tendsto (λ x, C + f x) l at_top) :
+  tendsto f l at_top :=
+(tendsto_at_top _ l).2 $ assume b,
+  monotone_mem_sets (λ x, le_of_add_le_add_left) ((tendsto_at_top _ _).1 hf (C + b))
+
+lemma tendsto_at_top_of_add_const_right (C : β) (hf : tendsto (λ x, f x + C) l at_top) :
+  tendsto f l at_top :=
+(tendsto_at_top _ l).2 $ assume b,
+  monotone_mem_sets (λ x, le_of_add_le_add_right) ((tendsto_at_top _ _).1 hf (b + C))
+
+lemma tendsto_at_top_of_add_bdd_above_left' :
+  (∃ C, {x | f x ≤ C} ∈ l) → tendsto (λ x, f x + g x) l at_top → tendsto g l at_top
```

Comment:
I would replace the existential quantifier in the assumption with an (explicit) variable `C` and an assumption `h : {x | f x ≤ C} ∈ l`.

Reply:
Done.